src/jit/vartype.h

```diff
@@ -276,6 +276,128 @@ inline bool varTypeIsEnregisterableStruct(T vt)
     return (TypeGet(vt) != TYP_STRUCT);
 }
 
+template <class T>
+inline bool varTypeUsesFloatReg(T vt)
+{
+    // Note that not all targets support SIMD, but if they don't, varTypeIsSIMD will
+    // always return false.
+    return varTypeIsFloating(vt) || varTypeIsSIMD(vt);
+}
+
+template <class T>
+inline bool varTypeUsesFloatArgReg(T vt)
+{
+#ifdef _TARGET_ARM64_
+    // Arm64 passes SIMD types in floating point registers.
+    return varTypeUsesFloatReg(vt);
+#else
+    // Other targets pass them as regular structs - by reference or by value.
+    return varTypeIsFloating(vt);
+#endif
+}
+
+//------------------------------------------------------------------------
+// varTypeIsValidHfaType: Determine if the type is a valid HFA type
+//
+// Arguments:
+//    vt - the type of interest
+//
+// Return Value:
+//    Returns true iff the type is a valid HFA type.
+//
+// Notes:
+//    This should only be called with the return value from GetHfaType().
+//    The only valid values are TYP_UNDEF, for which this returns false,
+//    TYP_FLOAT, TYP_DOUBLE, or (ARM64-only) TYP_SIMD*.
+//
+template <class T>
+inline bool varTypeIsValidHfaType(T vt)
+{
+#ifdef FEATURE_HFA
+    bool isValid = (TypeGet(vt) != TYP_UNDEF);
+    if (isValid)
+    {
+#ifdef _TARGET_ARM64_
+        assert(varTypeUsesFloatReg(vt));
+#else  // !_TARGET_ARM64_
+        assert(varTypeIsFloating(vt));
+#endif // !_TARGET_ARM64_
+    }
+    return isValid;
+#else  // !FEATURE_HFA
+    return false;
+#endif // !FEATURE_HFA
+}
+
+#ifdef FEATURE_HFA
+//------------------------------------------------------------------------
+// logSizeForHfaType: Get the logSize for an HFA type
+//
+// Arguments:
+//    vt - the type of interest
+//
+// Return Value:
+//    Returns base 2 log of the size of the type in 4-byte units.
+//
+// Notes:
+//    This should only be called with a valid HFA type.
+//    The only valid values are TYP_FLOAT, TYP_DOUBLE, or (ARM64-only) TYP_SIMD*.
+//
+template <class T>
+inline unsigned char logSizeForHfaType(T vt)
+{
+    unsigned char logSize = 0;
+    switch (vt)
```

Comment:
Here you probably want `switch (TypeGet(vt))` if you want the template to work as intended. Though I question the need for such templates...

Reply:
Yes, thanks! I don't have a strong opinion either way on the use of these templates, but since this seemed to belong in vartype.h, I went with the prevailing style there.